### PR TITLE
Add Bootstrap

### PR DESCRIPTION
**Pull request by topnimble**
#65 

## What happened

Add Bootstrap addon.

## Insight

1/ The Bootstrap addon depends on the DartSass.

2/ The Bootstrap addon supports both `Nimbe CSS, Nimble JS structure` or none.

3/ Rename the `css and js folder in the priv/template` to not make it confusing with the `bootstrap_css` 

image

4/ Refactor the template, only display the Nimble CSS, Boostrap prompt if the user choose DartSASS addon

image

## Proof Of Work

1/ The test is passed

2/ Boostrap is loaded

https://user-images.githubusercontent.com/11751745/156408715-b2cc94f2-13ac-4d29-8583-916c17a063f3.mov

3/ The project structure

image

image

### CODE REVIEW

**Review by andyduong1920 — APPROVED** [review]

File: lib/nimble_template/addons/variants/phoenix/web/bootstrap.ex [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        "dependencies": {
+          "@popperjs/core": "^2.11.2",
+          "bootstrap": "5.1.3",
+      """
+    )
+
+    project
+  end
+
+  def edit_app_js(project, %{with_nimble_js_addon: true}) do
```
Comment (longnd) [inline lib/nimble_template/addons/variants/phoenix/web/bootstrap.ex:45]: this method and the other ones, e.g. `edit_vendor_js` don't need to be public right?
```suggestion
  defp edit_app_js(project, %{with_nimble_js_addon: true}) do
```

File: lib/nimble_template/addons/variants/phoenix/web/bootstrap.ex [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    |> edit_files(opts)
+    |> copy_files()
+  end
+
+  defp edit_files(%Project{} = project, opts) do
+    project
+    |> edit_assets_package()
+    |> edit_app_js(opts)
+    |> edit_app_scss(opts)
+    |> edit_vendor_index(opts)
```
Comment (longnd) [inline lib/nimble_template/addons/variants/phoenix/web/bootstrap.ex:18]: we're missing the test for this behavior: import `bootstrap.scss` to `vendor/_index.scss`

**Review by longnd — COMMENTED** [review]
I recommend to take this chance to add some comments to `nimble_css/_variables.scss`
because we have seen many times in the previous projects, developers tend to mix the Boostrap variables and the app-specific variables. 
Adding 2 comments to separate the 2 sections can help avoid this case again.

```scss
////////////////////////////////
// Shared variables           //
////////////////////////////////


////////////////////////////////
// Custom Bootstrap variables //
////////////////////////////////
```

File: lib/nimble_template/addons/variants/phoenix/web/bootstrap.ex (hunk 1 above)
Comment (andyduong1920) [reply lib/nimble_template/addons/variants/phoenix/web/bootstrap.ex:45]: Good catch @longnd will update it

File: lib/nimble_template/addons/variants/phoenix/web/bootstrap.ex (hunk 2 above)
Comment (andyduong1920) [reply lib/nimble_template/addons/variants/phoenix/web/bootstrap.ex:18]: We have it @longnd 

image

File: priv/templates/nimble_template/assets/css_bootstrap/vendor/_bootstrap.scss [hunk 3]
```diff
@@ -0,0 +1,46 @@
+// Configuration
```
Comment (longnd) [inline priv/templates/nimble_template/assets/css_bootstrap/vendor/_bootstrap.scss:1]: image
I read the PR description, but I don't get the idea of why to name the directories in an inconsistency way:
`css_boostrap` vs `nimble_css/nimble_js`
could you clarify ? 🙏

Comment (andyduong1920) [reply priv/templates/nimble_template/assets/css_bootstrap/vendor/_bootstrap.scss:1]: css_bootstrap comes from @topnimble, and the `nimble_css and nimble_js` is just the addon.

If it's not clear, we can change to `bootstrap_css` :)

Comment (andyduong1920) [reply priv/templates/nimble_template/assets/css_bootstrap/vendor/_bootstrap.scss]: Changed to bootstrap_css in https://github.com/nimblehq/elixir-templates/pull/161/commits/5d289a05b1228ed3be5833de63b92e7adfdb773e

File: lib/nimble_template/addons/variants/phoenix/web/bootstrap.ex (hunk 1 above)
Comment (andyduong1920) [reply lib/nimble_template/addons/variants/phoenix/web/bootstrap.ex]: Updated it in https://github.com/nimblehq/elixir-templates/pull/161/commits/650f2461c2fac990450aacf24ea8650cd1ac0454

File: lib/nimble_template/addons/variants/phoenix/web/bootstrap.ex (hunk 2 above)
Comment (longnd) [reply lib/nimble_template/addons/variants/phoenix/web/bootstrap.ex:18]: that's right. My bad. I re-read the test 🙈

**Comment by andyduong1920** [comment]
> I recommend to take this chance to add some comments to `nimble_css/_variables.scss` because we have seen many times in the previous projects, developers tend to mix the Boostrap variables and the app-specific variables. Adding 2 comments to separate the 2 sections can help avoid this case again.
> 
> ```css-scss
> ////////////////////////////////
> // Shared variables           //
> ////////////////////////////////
> 
> 
> ////////////////////////////////
> // Custom Bootstrap variables //
> ////////////////////////////////
> ```

Sure @longnd I added that in https://github.com/nimblehq/elixir-templates/pull/161/commits/86e84f073a550bdb22faeac7e96b766cccb29d7d

**Review by longnd — APPROVED** [review]

**Pull request merged**